src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/OrganizationPaymentsReportingPagoPaRetrieverActivityImpl.java

```diff
@@ -0,0 +1,71 @@
+package it.gov.pagopa.payhub.activities.activity.paymentsreporting;
+
+import it.gov.pagopa.payhub.activities.connector.pagopapayments.PaymentsReportingPagoPaService;
+import it.gov.pagopa.payhub.activities.connector.processexecutions.IngestionFlowFileService;
+import it.gov.pagopa.pu.pagopapayments.dto.generated.PaymentsReportingIdDTO;
+import it.gov.pagopa.pu.processexecutions.dto.generated.IngestionFlowFile;
+import it.gov.pagopa.pu.processexecutions.dto.generated.IngestionFlowFile.FlowFileTypeEnum;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.time.OffsetDateTime;
+import java.util.Comparator;
+import java.util.List;
+
+@Lazy
+@Service
+public class OrganizationPaymentsReportingPagoPaRetrieverActivityImpl implements OrganizationPaymentsReportingPagoPaRetrieverActivity {
+	private static final FlowFileTypeEnum FLOW_FILE_TYPE = FlowFileTypeEnum.PAYMENTS_REPORTING_PAGOPA;
+
+	private final PaymentsReportingPagoPaService paymentsReportingPagoPaService;
+	private final IngestionFlowFileService ingestionFlowFileService;
+
+	public OrganizationPaymentsReportingPagoPaRetrieverActivityImpl(PaymentsReportingPagoPaService paymentsReportingPagoPaService,
+	                                                                IngestionFlowFileService ingestionFlowFileService) {
+		this.paymentsReportingPagoPaService = paymentsReportingPagoPaService;
+		this.ingestionFlowFileService = ingestionFlowFileService;
+	}
+
+	@Override
+	public List<Long> retrieve(Long organizationId) {
+		List<PaymentsReportingIdDTO> paymentsReportingIds = paymentsReportingPagoPaService.getPaymentsReportingList(organizationId);
+		OffsetDateTime oldestDate = findOldestDate(paymentsReportingIds);
```

Comment:
if null, there are not files to retrieve, you could already return an empty list instead of fetching ingestionFlowFiles